src/components/video_player.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * This component is mainly to handle the player's error and loading state nicely, being able to display something custom in those cases.
+ * The react-player component knows much more than youtube and these props, but only these are needed for now. Refer to it's documentation before changing this component.
+ */
+import * as React from 'react';
+import YouTubePlayer from 'react-player/lib/players/YouTube';
+
+import Spinner from "./spinner";
+
+export interface VideoPlayerProps {
+    /** url for the react-player component */
+    url: string;
+    /** Optional width to be given to the container and the video. Will accept values that are valid in CSS */
+    width?: string;
+    /** Optional height to be given to the container and the video. Will accept values that are valid in CSS */
+    height?: string;
+}
+
+export interface VideoPlayerState {
+    loading: boolean;
+    error: boolean;
+    wrongUrl: boolean;
+}
+
+export default class VideoPlayer extends React.Component<VideoPlayerProps, VideoPlayerState> {
+    static defaultProps = {
+        width: '100%',
+        height: '100%'
+    };
+
+    state = {
+        loading: true,
+        error: false,
+        wrongUrl: false
+    }
+
+    private _ready() {
+        this.setState({ loading: false })
+    }
+
+    private _error(e) {
+        this.setState({
+            loading: false,
+            error: true
+        });
+    }
+
+    private _playerClasses() {
+        const {loading,error,wrongUrl} = this.state;
+        if (loading || error || wrongUrl) {
+            return 'video-player--player video-player--player__hidden'
+        } else {
+            return 'video-player--player'
+        }
+    }
+
+    componentDidMount() {
+        if (!YouTubePlayer.canPlay(this.props.url)) {
+            this.setState({
+                loading: false,
+                wrongUrl: true
+            });
+        }
+    }
+
+    render() {
+        const { url, width, height } = this.props;
+        const { loading, wrongUrl, error } = this.state;
+
+        const playerProps = {
+            className: this._playerClasses(),
+            onReady: () => this._ready(),
+            onError: (e) => this._error(e),
+            url: url,
+            width: width ? width : '',
+            height: height ? height : '',
+        };
+        const containerStyle = {
+            width: width ? width : '',
+            height: height ? height : '',
+        }
+
+        return (
+            <div style={containerStyle} className='video-player'>
+                {wrongUrl && <span className='video-player--wrongurl'>Can not play Url</span>}
+                {error && <span className='video-player--error'>Video unavailable</span>}
+                {loading && <Spinner />}
+                <YouTubePlayer {...playerProps} />
```

Comment:
in the other components we list out the props on the component, rather than using a variable like this. It would be best to be consistent

Reply:
alright, changed it